### DISCUSSION TITLE
How to get the last uploaded file in a collection?

### DISCUSSION THREAD

**Discussion by MCKLtech** [spatie/laravel-medialibrary#2870] [Q&A] [answered]
My understanding is that **getFirstMedia** will return the first item in the Media Collection, but I assume this is not always the latest item added to that collection.

Is the filter variable documented anywhere? Or is there an out of the box function to get the latest uploaded file to a given collection?

**Comment by erikn69** [comment, marked as answer]
https://github.com/spatie/laravel-medialibrary/blob/92bc4b2d735b4de1753e5419c00ffc0060f560c4/src/InteractsWithMedia.php#L279-L284

Make your own trait, with your own logic
```php
public function getLastMedia(string $collectionName = 'default', $filters = []): ?Media 
 { 
     $media = $this->getMedia($collectionName, $filters); 
  
     return $media->last(); 
 } 
public function getLatestMedia(string $collectionName = 'default', $filters = []): ?Media 
 { 
     $media = $this->getMedia($collectionName, $filters); 
  
     return $media->sortBy('created_at')->values()->last(); 
 } 
```

**Reply by MCKLtech** [reply]
Oh my, that was obvious. What was I thinking! Thank you.

@freekmurze , might be a good addition to the library?
